commandapi-platforms/commandapi-velocity/commandapi-velocity-plugin/src/main/java/dev/jorel/commandapi/config/VelocityConfigurationAdapter.java

```diff
@@ -0,0 +1,105 @@
+package dev.jorel.commandapi.config;
+
+import org.spongepowered.configurate.CommentedConfigurationNode;
+import org.spongepowered.configurate.ConfigurationNode;
+import org.spongepowered.configurate.serialize.SerializationException;
+import org.spongepowered.configurate.yaml.YamlConfigurationLoader;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@SuppressWarnings("ConfusingArgumentToVarargsMethod")
+public record VelocityConfigurationAdapter(YamlConfigurationLoader loader, CommentedConfigurationNode config) implements ConfigurationAdapter<ConfigurationNode, DefaultedVelocityConfig> {
+
+	@Override
+	public void setValue(String key, Object value) {
+		setValue0(key, value);
+	}
+
+	@Override
+	public void setComment(String key, String[] comment) {
+		config.node(key.split("\\.")).comment(String.join("\n", comment));
+	}
+
+	@Override
+	public Object getValue(String key) {
+		return getValue0(key);
+	}
+
+	@SuppressWarnings("DataFlowIssue")
+	@Override
+	public String[] getComment(String key) {
+		return config.node(key.split("\\.")).comment().split("\n");
+	}
+
+	@Override
+	public Set<String> getKeys() {
+		Set<String> keys = new HashSet<>();
+		for (Object key : config.childrenMap().keySet()) {
+			keys.add((String) key);
+		}
+		return keys;
+	}
```

Comment:
I believe we could directly return the `keySet` to achieve the same result.

```suggestion
	@Override
	public Set<String> getKeys() {
		return config.childrenMap().keySet();
	}
```

However, this current implementation seems flawed. On Bukkit, `ConfigurationAdapter#getKeys` returns all the options including those inside a configuration section. It seems to me that `childrenMap` returns the base level of the config node structure, which in this case would incorrectly include the key for `messages` and omit the key for `messages.missing-executor-implementation`.